Dockerfile

```diff
@@ -599,17 +587,96 @@ RUN \
   mv cargo-deny-${DENYVER} cargo-deny
 
 WORKDIR /home/builder/cargo-deny
-COPY LICENSE-APACHE LICENSE-MIT /usr/share/licenses/cargo-deny/
+RUN cargo build --release --locked
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-cargo as sdk-rust-tools
+
+# Bring it all back together and run license-scan and cargo-deny on everything.
+
+COPY --from=sdk-cargo-deny \
+  /home/builder/cargo-deny \
+  /home/builder/cargo-deny
+
+COPY --from=sdk-license-tool \
+  /home/builder/license-tool \
+  /home/builder/license-tool
+
+COPY --from=sdk-license-scan \
+  /home/builder/license-scan \
+  /home/builder/license-scan
+
+COPY --chown=0:0 --from=sdk-cargo-deny \
+  /home/builder/cargo-deny/target/release/cargo-deny \
+  /usr/libexec/tools/
+
+COPY --chown=0:0 --from=sdk-license-tool \
+  /home/builder/license-tool/target/release/bottlerocket-license-tool \
+  /usr/libexec/tools/
+
+COPY --chown=0:0 --from=sdk-license-scan \
+  /home/builder/license-scan/target/release/bottlerocket-license-scan \
+  /usr/libexec/tools/
+
+COPY --chown=0:0 --from=sdk-license-scan \
+  /home/builder/license-scan/license-list-data/json/details \
+  /usr/libexec/tools/spdx-data
+
+COPY --chown=1000:1000 \
+  LICENSE-APACHE LICENSE-MIT \
+  /usr/share/licenses/cargo-deny/
+
+COPY --chown=1000:1000 \
+  COPYRIGHT LICENSE-APACHE LICENSE-MIT \
+  /usr/share/licenses/bottlerocket-license-tool/
+
+COPY --chown=1000:1000 \
+  COPYRIGHT LICENSE-APACHE LICENSE-MIT \
+  /usr/share/licenses/bottlerocket-license-scan/
```

Comment:
Shouldn't these also be owned by UID `0`? Aren't directories under `/usr/share/licenses/*` considered system directories?

Reply:
The permissions get fixed up later in the copy to the final image. Here we need the directories to be owned by `builder` in order for the license scan tool to write the attribution files for the vendored dependencies.